contrib/search-engine/packages.el

```diff
@@ -0,0 +1,82 @@
+;;; packages.el --- search-engine Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2014-2015 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;; List of all packages to install and/or initialize. Built-in packages
+;; which require an initialization must be listed explicitly in the list.
+(defvar search-engine-packages
+  '(
+    engine-mode
+    ))
+
+(defun search-engine/init-engine-mode ()
+  (use-package engine-mode
+    :init
+    (spacemacs/declare-prefix "a/" "Search Engines")
+    (evil-leader/set-key
+      "a/a" 'engine/search-amazon
+      "a/d" 'engine/search-duckduckgo
+      "a/g" 'engine/search-google
+      "a/i" 'engine/search-google-images
+      "a/G" 'engine/search-github
+      "a/m" 'engine/search-google-maps
+      "a/t" 'engine/search-twitter
+      "a/b" 'engine/search-project-gutemberg
+      "a/y" 'engine/search-youtube
+      "a/s" 'engine/search-stack-overflow
+      "a/S" 'engine/search-spacemacs-issues
+      "a/w" 'engine/search-wikipedia
+      "a/W" 'engine/search-wolfram-alpha
```

Comment:
For this kind of pattern consider using only the binding `SPC a /` and use an helm source to select the engine. An example is available here: https://github.com/syl20bnr/spacemacs/blob/develop/contrib/lang/sql/packages.el#L31-L46